### PR TITLE
Tree hashing changes

### PR DESCRIPTION
**Pull request by mikiquantum**
Tree Hashing:

- Leaf Node: sha3
- Internal Node: blake2b
- Modify tree.NewDefaultTree() with correspondent options.

### CODE REVIEW

**Comment by mikiquantum** [comment]
@vedhavyas the reduction in test coverage is caused by the fact that I disabled testworld until we have ur work in.

File: anchors/anchor_repository_integration_test.go [hunk 1]
```diff
@@ -18,6 +17,8 @@ import (
 	"github.com/centrifuge/go-centrifuge/utils"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/stretchr/testify/assert"
+
+	"golang.org/x/crypto/blake2s"
```
Comment (vedhavyas) [inline anchors/anchor_repository_integration_test.go]: is blake2s instead of blake2b intentional ?

File: crypto/generate.go [hunk 2]
```diff
@@ -39,10 +41,23 @@ func GenerateSigningKeyPair(publicFileName, privateFileName, curveType string) (
...
 
+// Blake2bHash wraps inconvenient blake2b_256 hashing ops
+func Blake2bHash(value []byte) (hash []byte, err error) {
+	h, err := blake2s.New256(nil)
+	if err != nil {
+		return nil, err
+	}
+	_, err = h.Write(value)
+	if err != nil {
+		return []byte{}, err
```
Comment (vedhavyas) [inline crypto/generate.go]:
```suggestion
		return nil, err
```

File: crypto/generate.go [hunk 3]
```diff
@@ -39,10 +41,23 @@ func GenerateSigningKeyPair(publicFileName, privateFileName, curveType string) (
...
 func GenerateHashPair(preimageSize int) (preimage, hash []byte, err error) {
 	preimage = utils.RandomSlice(preimageSize)
-	hash, err = Sha256Hash(preimage)
+	hash, err = Blake2bHash(preimage)
 	return preimage, hash, err
 }
 
+// Blake2bHash wraps inconvenient blake2b_256 hashing ops
+func Blake2bHash(value []byte) (hash []byte, err error) {
+	h, err := blake2s.New256(nil)
```
Comment (vedhavyas) [inline crypto/generate.go]: blake2b you mean ?

**Pull request merged**